From 9c11f10486021d217edec5424a6e201f05bc2efb Mon Sep 17 00:00:00 2001
From: Amir Ghanbari Bavarsad <amir.b@pervices.com>
Date: Sun, 20 Jan 2019 20:39:08 +0000
Subject: [PATCH 2/9] Fix DTS files for stratix 10

* Remove reserved memory that causes the kernel to freeze.
* Add SPI bus
* Disable I2C.
---
 .../boot/dts/altera/socfpga_stratix10.dtsi    | 44 +++++++++++--------
 .../dts/altera/socfpga_stratix10_socdk.dts    | 20 ++++++++-
 2 files changed, 44 insertions(+), 20 deletions(-)

diff --git a/arch/arm64/boot/dts/altera/socfpga_stratix10.dtsi b/arch/arm64/boot/dts/altera/socfpga_stratix10.dtsi
index adedd563125a..91dedd1b76fb 100644
--- a/arch/arm64/boot/dts/altera/socfpga_stratix10.dtsi
+++ b/arch/arm64/boot/dts/altera/socfpga_stratix10.dtsi
@@ -24,18 +24,18 @@
 	#address-cells = <2>;
 	#size-cells = <2>;
 
-	reserved-memory {
-		#address-cells = <2>;
-		#size-cells = <2>;
-		ranges;
-
-		service_reserved: svcbuffer@0 {
-                       compatible = "shared-dma-pool";
-                       reg = <0x0 0x0 0x0 0x1000000>;
-                       alignment = <0x1000>;
-                       no-map;
-               };
-	};
+	// reserved-memory {
+	// 	#address-cells = <2>;
+	// 	#size-cells = <2>;
+	// 	ranges;
+
+	// 	service_reserved: svcbuffer@0 {
+        //                compatible = "shared-dma-pool";
+        //                reg = <0x0 0x0 0x0 0x1000000>;
+        //                alignment = <0x1000>;
+        //                no-map;
+        //        };
+	// };
 
 	cpus {
 		#address-cells = <1>;
@@ -98,6 +98,12 @@
 		      <0x0 0xfffc6000 0x0 0x2000>;
 	};
 
+	spi_uart_clock: spi-uart-xtal {
+		#clock-cells = <0>;
+		compatible = "fixed-clock";
+		clock-frequency = <3686400>;
+	};
+
 	soc {
 		#address-cells = <1>;
 		#size-cells = <1>;
@@ -528,12 +534,12 @@
 			status = "disabled";
 		};
 
-		firmware {
-			svc {
-				compatible = "intel,stratix10-svc";
-				method = "smc";
-				memory-region = <&service_reserved>;
-			};
-		};
+		// firmware {
+		// 	svc {
+		// 		compatible = "intel,stratix10-svc";
+		// 		method = "smc";
+		// 		memory-region = <&service_reserved>;
+		// 	};
+		// };
 	};
 };
diff --git a/arch/arm64/boot/dts/altera/socfpga_stratix10_socdk.dts b/arch/arm64/boot/dts/altera/socfpga_stratix10_socdk.dts
index 2e3863ee12b3..d3f253d5d15c 100644
--- a/arch/arm64/boot/dts/altera/socfpga_stratix10_socdk.dts
+++ b/arch/arm64/boot/dts/altera/socfpga_stratix10_socdk.dts
@@ -125,7 +125,7 @@
 };
 
 &i2c1 {
-	status = "okay";
+	status = "disabled";
 	clock-frequency = <100000>;
 	i2c-sda-falling-time-ns = <890>;  /* hcnt */
 	i2c-sdl-falling-time-ns = <890>;  /* lcnt */
@@ -153,6 +153,24 @@
 	};
 };
 
+&spi0 {
+	status = "okay";
+
+	max14830@0 {
+		status = "okay";
+		compatible = "maxim,max14830";
+		reg = <0x0>;
+		spi-max-frequency = <1000000>;
+		clocks = <&spi_uart_clock>;
+		clock-names = "xtal";
+		interrupt-parent = <&intc>;
+		interrupts = <7 8>;
+		// gpio-controller;
+		// #gpio-cells = <2>;
+	};
+};
+ 
+
 &qspi {
 	flash@0 {
 		#address-cells = <1>;
-- 
2.19.1

